### PR TITLE
trmm: strided_batched and batched

### PR DESCRIPTION
**Pull request by amcamd**

### CODE REVIEW

File: library/src/blas3/rocblas_trmm.hpp [hunk 1]
```diff
@@ -122,15 +139,17 @@ rocblas_status rocblas_trmm_template(rocblas_handle    handle,
     //
     // -- Rewritten for gemm based trmm for rocBLAS
     //
-    T zero = 0.0;
-    T one  = 1.0;
+    T  zero   = 0.0;
```
Comment (daineAMD) [inline library/src/blas3/rocblas_trmm.hpp]: Didn't go through the original trmm algorithm, just went through the changes and quickly went through to ensure the indexes were translated right, looks fine as long as tests pass.

File: library/src/blas3/rocblas_trmm_strided_batched.cpp [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+                              n,
+                              LOG_BENCH_SCALAR_VALUE(alpha),
+                              "--lda",
+                              lda,
+                              "--stride_a",
+                              stride_a,
+                              "--ldb",
+                              ldc,
+                              "--stride_b",
+                              stride_c,
```
Comment (TorreZuk) [inline library/src/blas3/rocblas_trmm_strided_batched.cpp]: --stride_b but the interface is using stride_c.

File: library/src/blas3/rocblas_trmm_strided_batched.cpp [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+                            "n",
+                            n,
+                            "lda",
+                            lda,
+                            "stride_a",
+                            stride_a,
+                            "ldb",
+                            ldc,
+                            "stride_b",
+                            stride_c,
```
Comment (TorreZuk) [inline library/src/blas3/rocblas_trmm_strided_batched.cpp]: Not clear why a stride_c search and replace with stride_b wasn't done?     C API uses c but doxygen b.  I guess for consistency with original code this was done.

File: clients/include/testing_trmm_batched.hpp [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+                                    N,
+                                    &h_alpha[0],
+                                    dA.ptr_on_device(),
+                                    lda,
+                                    dB.ptr_on_device(),
+                                    ldb,
+                                    batch_count);
+        }
+        gpu_time_used  = get_time_us() - gpu_time_used;
+        rocblas_gflops = trmm_gflop_count<T>(M, N, side) * number_hot_calls / gpu_time_used * 1e6;
```
[the remark this answers is not in the document]
Comment (amcamd) [reply clients/include/testing_trmm_batched.hpp]: Thanks for catching,  fixed

File: clients/include/testing_trmm_strided_batched.hpp [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+    rocblas_int          K        = side == rocblas_side_left ? M : N;
+    const rocblas_stride stride_a = lda * K;
+    const rocblas_stride stride_b = ldb * N;
+    size_t               size_A   = batch_count * stride_a;
+    size_t               size_B   = batch_count * stride_b;
+
+    // allocate memory on device
+    device_vector<T> dA(size_A);
+    device_vector<T> dB(size_B);
+    if(!dA || !dB)
```
[the remark this answers is not in the document]
Comment (amcamd) [reply clients/include/testing_trmm_strided_batched.hpp]: Thanks for catching,  fixed

File: clients/include/testing_trmm_strided_batched.hpp [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+                                                          ldb,
+                                                          stride_b,
+                                                          batch_count),
+                          rocblas_status_invalid_handle);
+}
+
+template <typename T>
+void testing_trmm_strided_batched(const Arguments& arg)
+{
+    bool nantest = rocblas_isnan(arg.alpha) || rocblas_isnan(arg.alphai);
```
[the remark this answers is not in the document]
Comment (amcamd) [reply clients/include/testing_trmm_strided_batched.hpp]: Thanks, reversed comment out in testing_trmm_batched

File: clients/include/testing_trmm_strided_batched.hpp [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+    device_vector<T> dA(size_A);
+    device_vector<T> dB(size_B);
+    device_vector<T> alpha_d(1);
+
+    CHECK_HIP_ERROR(dA.memcheck());
+    CHECK_HIP_ERROR(dB.memcheck());
+    CHECK_HIP_ERROR(alpha_d.memcheck());
+
+    //  initialize full random matrix hA and hB
+    rocblas_init<T>(hA);
```
[the remark this answers is not in the document]
Comment (amcamd) [reply clients/include/testing_trmm_strided_batched.hpp]: Thanks, added rocblas_seedrand()

File: library/src/blas3/rocblas_trmm_batched.cpp [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+#include "rocblas.h"
+#include "rocblas_trmm.hpp"
+#include "utility.h"
+
+namespace
+{
+    template <typename>
+    constexpr char rocblas_trmm_batched_name[] = "unknown";
+    template <>
+    constexpr char rocblas_trmm_batched_name<float>[] = "rocblas_batched_strmm";
```
[the remark this answers is not in the document]
Comment (amcamd) [reply library/src/blas3/rocblas_trmm_batched.cpp]: Thanks, corrected

File: library/src/blas3/rocblas_trmm_strided_batched.cpp [hunk 9]
```diff
@@ -0,0 +1,373 @@
...
+#include "rocblas.h"
+#include "rocblas_trmm.hpp"
+#include "utility.h"
+
+namespace
+{
+    template <typename>
+    constexpr char rocblas_trmm_strided_batched_name[] = "unknown";
+    template <>
+    constexpr char rocblas_trmm_strided_batched_name<float>[] = "rocblas_strided_batched_strmm";
```
[the remark this answers is not in the document]
Comment (amcamd) [reply library/src/blas3/rocblas_trmm_strided_batched.cpp]: Thanks,  corrected

File: clients/include/testing_trmm_strided_batched.hpp [hunk 10]
```diff
@@ -0,0 +1,387 @@
...
+                                                            &h_alpha_T,
+                                                            dA,
+                                                            lda,
+                                                            stride_a,
+                                                            dB,
+                                                            ldb,
+                                                            stride_b,
+                                                            batch_count));
+
+        CHECK_HIP_ERROR(hipMemcpy(hB_1, dB, sizeof(T) * size_B, hipMemcpyDeviceToHost));
```
[the remark this answers is not in the document]
Comment (amcamd) [reply clients/include/testing_trmm_strided_batched.hpp]: Thanks for catching, changed to use helpers.

File: library/src/blas3/rocblas_trmm_strided_batched.cpp (hunk 2 above)
Comment (amcamd) [reply library/src/blas3/rocblas_trmm_strided_batched.cpp]: Changed to consistently use b, ldb, stride_b

File: library/src/blas3/rocblas_trmm_strided_batched.cpp (hunk 3 above)
Comment (amcamd) [reply library/src/blas3/rocblas_trmm_strided_batched.cpp]: change to consistently use b, ldb, stride_b in place of sometimes c, ldc, stride_c

**Comment by amcamd** [comment]
All tests pass in CI, and run independently on standalone machines.

**Pull request merged**